openapi/api.yml

```diff
@@ -0,0 +1,204 @@
+openapi: "3.0.0"
+
+info:
+  description: Fund store API for DLUHC Funding Service Design
+  version: "0.1.0"
+  title: Funding Service Design - Fund store.
+
+paths:
+  /funds:
+    get:
+      tags:
+      - Funds
+      summary: Returns list of all funds
+      description: Returns list of all funds
+      operationId: core.funds.get_funds
+      parameters:
+        - in: query
+          name: search_items
+          style: form
+          schema:
+            type: array
+            items: 
+              type: string
+          required: false
+          explode: false
+      responses:
+        200:
+          description: "List all funds."
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref : '#/components/schemas/fund'
+        404:
+          description: "funds page not found."
+          content:
+              texts/plain:
+                schema:
+                  title: Funds page not found
+                  type: string
+                  example: Not found
+
+  /funds/{fund_id}/rounds/{round_id}:
+    get:
+      tags:
+      - Rounds
+      summary: Returns the data on a specified round for a specific fund.
+      description: Given a fund ID and a round ID we return the relavant round data.
+      operationId: core.rounds.get_round
+      parameters:
+        - in: path
+          name: fund_id
+          schema:
+            type: string
+          required: true
+        - in: path
+          name: round_id
+          schema:
+            type: string
+          required: true
+      responses:
+        200:
+          description: "If the round exists then the data is returned."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/round'
+        404:
+          description: "Round not found from given fund id and round id."
+          content:
+              texts/plain:
+                schema:
+                  title: Round not found
+                  type: string
+                  example: Not found
+
+  /funds/{fund_id}:
+    get:
+      tags:
+      - Funds
+      summary: Returns the data on a specified fund.
+      description: Given a fund ID we return the relevant fund data.
+      operationId: core.funds.get_fund
+      parameters:
+        - in: path
+          name: fund_id
+          schema:
+            type: string
+          required: true
+      responses:
+        200: 
+          description: "If the fund exists then the data is returned."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/fund'
+        404:
+          description: "Fund page not found for given fund id."
+          content:
+              texts/plain:
```

Comment:
Do we have a consistent approach with 404 response types across our apis? Perhaps not yet, but would be good to agree one - I know some 400~ responses return a json error (rather than text/plain). Could it be useful to have a json response with an error message that can be from it here? To discuss..!

Reply:
👌